### PR TITLE
Added custom options and ability to set payment method/token in dummy order creation

### PR DESCRIPTION
**Pull request by wsheldon**
Hey Kalen,

We needed these customizations for a project I'm working on.  Feel free to merge if you find them useful.

The payment method/vault token works with Braintree but would likely work for others, I presume.  

Usage is:

mr order:create:dummy --custom_options="Customname,Single" --payment="braintree" --cc_token="######" 1

Thanks,
Wilson

### CODE REVIEW

**Comment by wsheldon** [comment]
Btw, have you figured out a way to trigger xDebug breakpoints when working in Magerun?

**Comment by kalenjordan** [comment]
Nice I'll try to take a look at that next week.  I think xdebug usually works if I run it with the same php instance that is configured to use xdebug over my local apache (which is sometimes *not* what is aliased to `php` in the command line).

File: src/KJ/Magento/Command/Order/Create/DummyCommand.php [hunk 1]
```diff
@@ -40,6 +40,9 @@ protected function configure()
             ->addOption('product', null, InputOption::VALUE_OPTIONAL, "A product SKU to use for the order")
             ->addOption('store', null, InputOption::VALUE_OPTIONAL, "A store ID to use for the order")
             ->addOption('shipping', null, InputOption::VALUE_OPTIONAL, "A shipping method code to use for the order")
+			->addOption('custom_options', null, InputOption::VALUE_OPTIONAL, "Custom options to use for the product, if applicable")
```
Comment (kalenjordan) [inline src/KJ/Magento/Command/Order/Create/DummyCommand.php]: @wsheldon looks like there's an indentation / spacing issue here and throughout the PR.

File: src/KJ/Magento/Command/Order/Create/DummyCommand.php [hunk 2]
```diff
@@ -245,8 +248,20 @@ protected function addItemToQuote()
         $product = $this->getProduct();
         $quote = $this->getQuote();
 
+		$request = null;
```
Comment (kalenjordan) [inline src/KJ/Magento/Command/Order/Create/DummyCommand.php]: Nice

**Comment by kalenjordan** [comment]
This is great thanks man.  If you could just tighten up that indentation issue we should be good to go.

File: src/KJ/Magento/Command/Order/Create/DummyCommand.php (hunk 1 above)
Comment (wsheldon) [inline src/KJ/Magento/Command/Order/Create/DummyCommand.php]: Hmm.  I'll check.  I thought PHPStorm would match your existing spacing.

**Comment by wsheldon** [comment]
Ok, check now.  Needed to re-enable 'Smart Tabs' in PHPStorm.

**Pull request merged**